### PR TITLE
switch to uvicorn for tiler api

### PR DESCRIPTION
**Pull request by vincentsarago**
## Description

This PR switch Gunicorn to Uvicorn. This can help with memory management. 

> In particular, when running on Kubernetes you will probably not want to use Gunicorn and instead run a single Uvicorn process per container

ref https://fastapi.tiangolo.com/deployment/server-workers/#server-workers-gunicorn-with-uvicorn

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review
- [ ] Changelog has been updated
- [ ] Documentation has been updated
- [ ] Unit tests pass locally (./scripts/test)
- [ ] Code is linted and styled (./scripts/format)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -119,7 +119,6 @@ services:
       - DB_MIN_CONN_SIZE=1
       - DB_MAX_CONN_SIZE=1
       - WEB_CONCURRENCY=1
```
Comment (vincentsarago) [inline docker-compose.yml:121]: Only Web_concurrency is used in Uvicorn

File: pctiler/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.8.7-buster
+FROM python:3.9-slim
```
Comment (vincentsarago) [inline pctiler/Dockerfile:1]: python 3.9 add better threads management/definition (doesn't hurt IMO)

File: pctiler/Dockerfile [hunk 3]
```diff
@@ -11,24 +11,15 @@ ARG DEVOPS_PAT
...
-COPY pctiler/gunicorn_conf.py gunicorn_conf.py
-
 COPY pccommon /opt/src/pccommon
 COPY pctiler /opt/src/pctiler
 RUN pip install -e ./pccommon -e ./pctiler[server]
 
 # GDAL config
 ENV GDAL_CACHEMAX 200
-ENV GDAL_INGESTED_BYTES_AT_OPEN 16383
+ENV GDAL_INGESTED_BYTES_AT_OPEN 32768
```
Comment (vincentsarago) [inline pctiler/Dockerfile:22]: Double byte at ingest, could help when reading big COGs

File: pctiler/Dockerfile [hunk 4]
```diff
@@ -37,14 +28,12 @@ ENV GDAL_HTTP_VERSION 2
 # https://github.com/rasterio/rasterio/issues/2415
 ENV GTIFF_REPORT_COMPD_CS=0
 ENV VSI_CACHE TRUE
-ENV VSI_CACHE_SIZE 5000000
+ENV VSI_CACHE_SIZE 50000000
```
Comment (vincentsarago) [inline pctiler/Dockerfile:31]: value is in bytes, IMO 5000000 (5Mb) is better then 0.5 Mb

File: pctiler/Dockerfile [hunk 5]
```diff
@@ -37,14 +28,12 @@ ENV GDAL_HTTP_VERSION 2
...
+ENV CPL_VSIL_CURL_CACHE_SIZE 200000000
```
Comment (vincentsarago) [inline pctiler/Dockerfile:32]: I'm not 💯 sure this change something but worth a try

**Review by mmcfarland — APPROVED** [review]
I haven't been able to reproduce memory issues locally that we see in prod, but based on the fastapi docs, these seem like positive changes regardless. 👍

**Pull request merged**